src/v/pandaproxy/rest/handlers.cc

```diff
@@ -327,8 +326,13 @@ create_consumer(server::request_t rq, server::reply_t rp) {
         co_return std::move(rp);
     };
 
-    co_return co_await rq.service().client().invoke_on(
-      group_shard, rq.context().smp_sg, std::move(handler));
+    co_return co_await rq.service().client_cache().invoke_on_cache(
+      group_shard,
+      rq.context().smp_sg,
+      [user{rq.user}, authn_method{rq.authn_method}, h{std::move(handler)}](
+        kafka_client_cache& cache) mutable -> ss::future<server::reply_t> {
+          return h(cache.fetch_or_insert(user, authn_method));
+      });
```

Comment:
I think it might be preferable to dispatch the consumer work on the shard for the user.

I think the benefits of keeping consumer group stuff on the same shard/in the same client are overshadowed by the losses of creating a new client for a user on another shard, perhaps just to find a coordinator.